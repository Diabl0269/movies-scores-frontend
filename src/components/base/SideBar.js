import React from 'react'
import {
  Slide,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  makeStyles,
  Drawer
} from '@material-ui/core'
import HomeIcon from '@material-ui/icons/Home'
import AccountCircleIcon from '@material-ui/icons/AccountCircle'
import d from '../../data/dictionary'
import { navigate } from 'hookrouter'

const useStyles = makeStyles({
  list: {
    width: 250
  }
})

export default ({ open, setOpen }) => {
  const classes = useStyles()

  const links = [
    { text: d('home'), icon: <HomeIcon />, clickHandler: () => navigate('/') },
    { text: d('account'), icon: <AccountCircleIcon />, clickHandler: () => navigate('/account') }
  ]

  const toggleDrawer = (event) => {
    if (event && event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
      return
    }

    setOpen(!open)
  }

  return (
    <Drawer open={open} onClose={toggleDrawer}>
      <Slide direction="right" in={open} mountOnEnter unmountOnExit>
        <div className={classes.list} onClick={toggleDrawer} onKeyDown={toggleDrawer}>
          <List>
            {links.map(({ text, icon, clickHandler }) => (
              <ListItem button key={text} onClick={clickHandler}>
                <ListItemIcon>{icon}</ListItemIcon>
                <ListItemText primary={text} />
              </ListItem>
            ))}
          </List>
        </div>
      </Slide>
    </Drawer>
  )
}
