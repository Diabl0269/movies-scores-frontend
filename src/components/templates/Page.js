import React, { useState } from 'react'
import SideBar from '../base/SideBar'
import Header from '../base/Header'
import Footer from '../base/Footer'

export default ({Content, title}) => {
  const [open, setOpen] = useState(false)

  return (
    <>
      {open && <SideBar open={open} setOpen={setOpen} />}
      <div id='innerPageContainer'>
        <Header setOpen={setOpen} title={title} />
        <Content />
        <Footer />
      </div>
    </>
  )
}
