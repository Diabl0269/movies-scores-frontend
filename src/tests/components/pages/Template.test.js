import React from 'react'
import { mount, shallow } from 'enzyme'
import { IconButton } from '@material-ui/core'
import Page from '../../../components/pages/Page'
import SideBar from '../../../components/base/SideBar'

it('should match snapshot', () => {
  const wrapper = shallow(<Page />)
  expect(wrapper.debug()).toMatchSnapshot()
})

it('should match snapshot with open sidebar', () => {
  const wrapper = mount(<Page />)
  wrapper.find(IconButton).at(0).simulate('click')
  expect(wrapper.find(SideBar).exists()).toBeTruthy()
})
