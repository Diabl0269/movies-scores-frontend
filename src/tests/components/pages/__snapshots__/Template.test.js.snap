// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
"<Fragment>
  <div id=\\"innerPageContainer\\">
    <_default setOpen={[Function: bound _dispatchAction]} title=\\"title\\" />
    <_default />
  </div>
</Fragment>"
`;
